File: src/main/java/um/haberes/core/model/view/CodigoSearch.java
package um.haberes.core.model.view;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import org.hibernate.annotations.Immutable;
import um.haberes.core.model.AuditableEntity;

@Entity
@Table(name = "vw_codigo_search")
@Immutable
@Getter
@Setter
@EqualsAndHashCode(callSuper = false)
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class CodigoSearch extends AuditableEntity {

    @Id
    private Integer codigoId = null;

    private String nombre = null;

    private Byte docente = null;

    private Byte noDocente = null;

    private String search = null;
}

File: src/main/java/um/haberes/core/kotlin/model/view/CodigoSearch.kt
package um.haberes.core.kotlin.model.view

import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.hibernate.annotations.Immutable
import um.haberes.core.kotlin.model.Auditable

@Entity
@Table(name = "vw_codigo_search")
@Immutable
data class CodigoSearch(

    @Id
    var codigoId: Int? = null,
    var nombre: String? = null,
    var docente: Byte? = null,
    var noDocente: Byte? = null,
    var search: String? = null

) : Auditable()
